### ISSUE TITLE
S40RTS data directory assumes "/" at end

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#983]
just add it if it is not there. @bjs2

**Comment by bjs2** [comment]
I'm actually not sure where to edit this in the code. I'm still running the preconditioner tests, so it's not really urgent.

**Comment by bangerth** [comment]
The issue is in S40RTS.cc, line 161:
```
spherical_harmonics_lookup
  .reset(new internal::S40RTS::SphericalHarmonicsLookup(datadirectory+harmonics_coeffs_file_name,
                                                                                               this->get_mpi_communicator()));
```
For this to work, `datadirectory` needs to be terminated with a slash. The default value of the corresponding parameter is in fact so terminated, but that need not be so if the user does not provide a slash at the end of a user-specified directory, because we have this code in line 363:
```
          datadirectory = Utilities::replace_in_string(prm.get ("Data directory"),
                                                       "$ASPECT_SOURCE_DIR",
                                                       ASPECT_SOURCE_DIR);
```
This just need the extra statement
```
  if ((datadirectory.size()>0) && (datadirectory[data_directory.size()-1] != '/'))
    datadirectory += "/";
```

Bonus points if you rename `datadirectory` to `data_directory`. Extra bonus points if you check whether the same problem exists in the `SAVANI.cc` file.

**Comment by gassmoeller** [comment]
Fixed by #1021
